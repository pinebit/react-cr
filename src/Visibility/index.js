import React from 'react';
import renderChildren from '../utils/renderChildren';

const Visibility = (props) => {
  const { children, visible, ...otherProps } = props;
  return visible ? renderChildren(children, otherProps) : null;
};

Visibility.propTypes = {
  visible: React.PropTypes.bool
};

export default Visibility;
