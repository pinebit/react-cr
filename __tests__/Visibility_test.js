import React from 'react';
import renderer from 'react-test-renderer';
import Visibility from '../src/Visibility';

it('Visibility renders correctly when visible is false', () => {
  const tree = renderer.create(
    <Visibility visible={false}><h1>Test</h1></Visibility>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Visibility renders correctly when visible is true', () => {
  const tree = renderer.create(
    <Visibility visible><h1>Test</h1></Visibility>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
