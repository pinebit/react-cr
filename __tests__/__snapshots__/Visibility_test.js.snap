// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visibility renders correctly when visible is false 1`] = `null`;

exports[`Visibility renders correctly when visible is true 1`] = `
<h1>
  Test
</h1>
`;
